cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/default-orchestrator/DefaultOrchestrator.tsx

```diff
@@ -0,0 +1,38 @@
+import { ResourceAction, StoreDispatch } from '../../../shared/types/types';
+import { RESOURCE } from '../../../state-management/actions/Resources.actions';
+import { AWS_FIELDS } from '../aws/aws-mc-basic/AwsManagementClusterBasic.constants';
+import { AZURE_FIELDS } from '../azure/azure-mc-basic/AzureManagementClusterBasic.constants';
+
+export function clearPreviousResourceData(dispatch: StoreDispatch, actionType: RESOURCE, resourceName: AWS_FIELDS | AZURE_FIELDS) {
+    dispatch({
+        type: actionType,
+        resourceName: resourceName,
+        payload: [],
+    } as ResourceAction);
+}
+
+export function saveCurrentResourceData(
+    dispatch: StoreDispatch,
+    actionType: RESOURCE,
+    resourceName: AWS_FIELDS | AZURE_FIELDS,
```

Comment:
Would you consider a simple `string` for the resourceName type?

Reply:
Yes sir. That would be more simple. That would be a good approach to deal with those enum type (`AWS_FIELDS` and `AZURE_FIELDS`)